pkg/ddl/placement_policy.go

```diff
@@ -367,17 +367,17 @@ func CheckPlacementPolicyNotInUseFromInfoSchema(is infoschema.InfoSchema, policy
 		if ref := dbInfo.PlacementPolicyRef; ref != nil && ref.ID == policy.ID {
 			return dbterror.ErrPlacementPolicyInUse.GenWithStackByArgs(policy.Name)
 		}
+	}
 
-		tblInfos, err := is.SchemaTableInfos(context.Background(), dbInfo.Name)
-		if err != nil {
-			return errors.Trace(err)
-		}
-		for _, tblInfo := range tblInfos {
+	schemaTables := is.ListTablesWithSpecialAttribute(infoschema.AllPlacementPolicyAttribute)
+	for _, schemaTable := range schemaTables {
```

Comment:
Should we filter with `dbInfo.Name` here?

Reply:
Here just retrieve and check all tables with necessary attributes, so it's seperated from the previous loop.